spec/ProtectedFields.spec.js

```diff
@@ -138,4 +141,246 @@ describe('ProtectedFields', function() {
+          readUserFields: ['owner'],
+          protectedFields: { '*': ['owner'], 'readUserFields:owner': [] },
+        }
+      );
+
+      await Parse.User.logIn('user1', 'password');
+      const objectAgain = await obj.fetch();
+      expect(objectAgain.get('owner').id).toBe(user1.id);
+      expect(objectAgain.get('test')).toBe('test');
+      done();
+    });
+
+    it('should deny access to other users using single user pointer-permissions', async done => {
+      const config = Config.get(Parse.applicationId);
+      const obj = new Parse.Object('AnObject');
+
+      obj.set('owner', user1);
+      obj.set('test', 'test');
+      await obj.save();
+
+      const schema = await config.database.loadSchema();
+      await schema.updateClass(
+        'AnObject',
+        {},
+        {
+          get: { '*': true },
+          find: { '*': true },
+          readUserFields: ['owner'],
+          protectedFields: { '*': ['owner'], 'readUserFields:owner': [] },
+        }
+      );
+
+      await Parse.User.logIn('user2', 'password');
+      const objectAgain = await obj.fetch();
+      expect(objectAgain.get('owner')).toBe(undefined);
+      expect(objectAgain.get('test')).toBe('test');
+      done();
+    });
+
+    it('should deny access to public using single user pointer-permissions', async done => {
+      const config = Config.get(Parse.applicationId);
+      const obj = new Parse.Object('AnObject');
+
+      obj.set('owner', user1);
+      obj.set('test', 'test');
+      await obj.save();
+
+      const schema = await config.database.loadSchema();
+      await schema.updateClass(
+        'AnObject',
+        {},
+        {
+          get: { '*': true },
+          find: { '*': true },
+          readUserFields: ['owner'],
+          protectedFields: { '*': ['owner'], 'readUserFields:owner': [] },
+        }
+      );
+
+      const objectAgain = await obj.fetch();
+      expect(objectAgain.get('owner')).toBe(undefined);
+      expect(objectAgain.get('test')).toBe('test');
+      await Parse.User.logIn('user1', 'password');
+      done();
+    });
+
+    it('should allow access using user array pointer-permissions', async done => {
+      const config = Config.get(Parse.applicationId);
+      const obj = new Parse.Object('AnObject');
+
+      obj.set('owners', [user1, user2]);
+      obj.set('test', 'test');
+      await obj.save();
+
+      const schema = await config.database.loadSchema();
+      await schema.updateClass(
+        'AnObject',
+        {},
+        {
+          get: { '*': true },
+          find: { '*': true },
+          readUserFields: ['owners'],
+          protectedFields: { '*': ['owners'], 'readUserFields:owners': [] },
+        }
+      );
+
+      await Parse.User.logIn('user1', 'password');
+      let objectAgain = await obj.fetch();
+      expect(objectAgain.get('owners')[0].id).toBe(user1.id);
+      expect(objectAgain.get('test')).toBe('test');
+      await Parse.User.logIn('user1', 'password');
```

Comment:
Why are you logging in again the same user? Should be `user2`?

Reply:
Correct. I'll fix the typo.

Reply:
Should now be fixed by [this commit](https://github.com/parse-community/parse-server/pull/5951/commits/07388b8caad1f9711d4035a7d317d33e7d0af775)